bundles/http_admin/http_admin/src/service_tree.c

```diff
@@ -307,8 +307,8 @@ service_tree_node_t *findServiceNodeInTree(service_tree_t *svc_tree, const char
 
         char *uri_token = strtok_r(uri_cpy, "/", &save_ptr);
         //Check for horizontal matches for the first token
-        while(current != NULL) {
-            if(strcmp(current->svc_data->sub_uri, uri_token) == 0){
+        while (current != NULL) {
+            if (uri_token != NULL && strcmp(current->svc_data->sub_uri, uri_token) == 0){
```

Comment:
I think if (uri_token == NULL) a deadlock will occur, you have to catch this case too. The uri_token will stay NULL and the current pointer will not change to NULL, it should be enough to change the current to NULL.

Reply:
added a check for uri_token == NULL